pkg/agent/run.go

```diff
@@ -64,6 +69,14 @@ func Run(cmd *cobra.Command, args []string) {
 	defer cancel()
 	config, preflightClient := getConfiguration()
 
+	if Profiling {
+		log.Printf("pprof profiling was enabled.\nRunning profiling on port :6060")
+		go func() {
+			err := http.ListenAndServe(":6060", nil)
+			log.Fatalf("failed to run pprof profiler: %s", err)
```

Comment:
This will exit the application regardless of the error that `ListenAndServe` returns, we shouldn't call `Fatal` for a `nil` error or for `http.ErrServerStopped` (I think that's what the error is called)